internal/publish/pull_requests.go

```diff
@@ -0,0 +1,116 @@
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func openPullRequest(githubClient *github.Client, githubUser, destinationBranch string, manifest packages.PackageManifest, commitHash string) error {
+	title := buildPullRequestTitle(manifest)
+	diffURL := buildPullRequestDiffURL(githubUser, commitHash)
+	description := buildPullRequestDescription(manifest, diffURL)
+
+	userHead := fmt.Sprintf("%s:%s", githubUser, destinationBranch)
+	maintainerCanModify := true
+	base := snapshotStage
+
+	logger.Debugf("Create new pull request (head: %s, base: %s)", userHead, base)
+	pullRequest, _, err := githubClient.PullRequests.Create(context.Background(), repositoryOwner, repositoryName, &github.NewPullRequest{
+		Title:               &title,
+		Head:                &userHead,
+		Base:                &base,
+		Body:                &description,
+		MaintainerCanModify: &maintainerCanModify,
+	})
+	if err != nil {
+		return errors.Wrap(err, "can't open new pull request")
+	}
+
+	logger.Debugf("Pull request URL: %s", *pullRequest.HTMLURL)
+
+	// Try to set reviewers
+	reviewers := buildReviewersRequest(manifest)
+	logger.Debugf("Update reviewers (pull request ID: %d)", *pullRequest.Number)
+	pullRequest, _, err = githubClient.PullRequests.RequestReviewers(context.Background(), repositoryOwner, repositoryName, *pullRequest.Number, reviewers)
+	if err != nil {
+		return errors.Wrap(err, "can't request reviewers, please double-check if the owner exists and has access to the repository")
+	}
+
+	if len(pullRequest.RequestedTeams) != 0 || len(pullRequest.RequestedReviewers) != 0 {
+		logger.Debugf("Reviewers requested successfully (teams: %d, reviewers: %d)", len(pullRequest.RequestedTeams), len(pullRequest.RequestedReviewers))
+		return nil
+	}
+
+	// Fallback reviewers to default package owner
+	logger.Debugf("Update reviewers with default owner (pull request ID: %d)", *pullRequest.Number)
+	pullRequest, _, err = githubClient.PullRequests.RequestReviewers(context.Background(), repositoryOwner, repositoryName, *pullRequest.Number,
+		buildDefaultReviewersRequest())
+	if err != nil {
+		return errors.Wrap(err, "can't request reviewers with default owner")
+	}
+	return nil
+}
+
+func buildPullRequestTitle(manifest packages.PackageManifest) string {
+	return fmt.Sprintf(`[snapshot] Update "%s" integration to version %s`, manifest.Name, manifest.Version)
+}
+
+func buildPullRequestDiffURL(username, commitHash string) string {
+	return fmt.Sprintf("https://github.com/%s/package-storage/commit/%s", username, commitHash)
+}
+
+func buildPullRequestDescription(manifest packages.PackageManifest, diffURL string) string {
+	return fmt.Sprintf("This PR updates `%s` integration to version %s.\n\nChanges: %s", manifest.Name,
+		manifest.Version, diffURL)
+}
+
+func buildReviewersRequest(manifest packages.PackageManifest) github.ReviewersRequest {
+	if manifest.Owner.Github == "" {
+		return buildDefaultReviewersRequest()
+	}
```

Comment:
I'm torn on whether we should default the reviewer to `elastic/integrations`, especially given that `owner.github` is [required](https://github.com/elastic/package-spec/blob/a84a74e8d92bdaab5fa5f07841fbe786c1bed14b/versions/1/manifest.spec.yml#L304) by the spec. OTOH, we could some day change the spec to not require a github team owner and then this default might prove handy.

Reply:
Actually it's a different reason. If the tool can't set the reviewers to the original owner (has been deleted from Github settings, internal issue?), the tool will fallback to "elastic/integrations", so we can track it.

Reply:
... until one day `elastic/integrations` is also deleted. 😄 😬 😜